src/routers/transactions.py

```diff
@@ -37,8 +39,42 @@ def create_transaction(transaction: TransactionCreate, session: SessionDep):
 
 
 @router.get("/", response_model=list[TransactionRead])
-def read_transactions(session: SessionDep):
-    transactions = session.exec(select(Transaction)).all()
+def read_transactions(
+    session: SessionDep,
+    pagination_input: PaginationDep,
+    q: Annotated[str | None, Query(max_length=40)] = None,
+):
+    query = select(Transaction)
```

Comment:
checking for n+1 issues, all good, I think the fact you have `category: Category | None = Relationship(..., sa_relationship_kwargs={"lazy": "selectin"})` we won't have redundant/extra queries here 📈

Reply:
not sqlmodel but still relevant for the concept:

[video] Optimizing Django Queries: How to Solve the N+1 Problem with select_related()
https://www.youtube.com/watch?v=e_8JvcP1q48